cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -361,4 +361,74 @@ metadata:
   name: coredns
   namespace: kube-system
 `
+
+	// CoreDNSCacheDaemonSet is the CoreDNS cache DaemonSet manifest
+	CoreDNSCacheDaemonSet = `
+  apiVersion: apps/v1
+  kind: DaemonSet
+  metadata:
+    name: {{ .DeploymentName }}
+    namespace: kube-system
+    labels:
+      k8s-app: kube-dns
+      kubernetes.io/cluster-service: "true"
+      addonmanager.kubernetes.io/mode: Reconcile
+  spec:
+    selector:
+      matchLabels:
+        k8s-app: node-local-dns
+    template:
+      metadata:
+         labels:
+            k8s-app: node-local-dns
+      spec:
+        priorityClassName: system-node-critical
+        serviceAccountName: node-local-dns
+        hostNetwork: true
+        dnsPolicy: Default  # Don't use cluster DNS.
+        tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+        - key: {{ .MasterTaintKey }}
+          effect: NoSchedule
+        containers:
+        - name: node-cache
+          image: {{ .Image }}
+          resources:
+            limits:
+              memory: 30Mi
+            requests:
+              cpu: 25m
+              memory: 5Mi
```

Comment:
what are the guaranties for these system resources in terms of DNS cache?
are these recommended settings by a page somewhere?

Reply:
That's the resources that were used for the benchmark and recommended here: https://github.com/kubernetes/community/blob/master/keps/sig-network/0030-nodelocal-dns-cache.md#choice-of-caching-agent

Reply:
@srueg ^^

Reply:
It's using the CoreDNS cache plugin:
> Each element in the cache is cached according to its TTL (with TTL as the max). A cache is divided into 256 shards, each holding up to 39 items by default - for a total size of 256 * 39 = 9984 items.

https://github.com/coredns/coredns/tree/master/plugin/cache#syntax